pkgs/by-name/co/compiler-explorer-cpp/package.nix

```diff
@@ -0,0 +1,204 @@
+                name = "clang-query";
+                type = "independent";
+                class = "clang-query-tool";
+                stdinHint = "Query commands";
+              };
+
+            lld = {
+              name = "ldd";
+              exe = "${lld_17}/bin/lld_17";
+              type = "postcompilation";
+              class = "readelf-tool";
+              exclude = "djggp";
+              stdinHint = "disabled";
+            };
+
+            readelf =
+              {
+                name = "readelf";
+                exe = "${defaultGcc}/bin/readelf";
+                type = "postcompilation";
+                class = "readelf-tool";
+                exclude = "djggp";
+                stdinHint = "disabled";
+              };
+
+            nm =
+              {
+                name = "nm";
+                exe = "${defaultGcc}/bin/nm";
+                type = "postcompilation";
+                class = "nm-tool";
+                exclude = "djggp";
+                stdinHint = "disabled";
+              };
+            strings =
+              {
+                name = "strings";
+                exe = "${defaultGcc}/bin/strings";
+                type = "postcompilation";
+                class = "strings-tool";
+                exclude = "djggp";
+                stdinHint = "disabled";
+              };
+
+            llvmdwarfdumpdefault =
+              {
+                name = "llvm-dwarfdump";
+                exe = "${llvmPackages_17.bintools-unwrapped}/bin/llvm-dwarfdump";
+                type = "postcompilation";
+                class = "llvm-dwarfdump-tool";
+                stdinHint = "disabled";
+              };
+          };
+
+
+        defaultCompiler = "${(compilerEntry defaultGcc "gcc" gccCommand).entry}";
+        postProcess = "";
+        demangler = "${defaultGcc}/bin/c++filt";
+        demanglerType = "cpp";
+        objdumper = "${defaultGcc}/bin/objdump";
+        options = "";
+        supportsBinary = "true";
+        supportsBinaryObject = "true";
+        binaryHideFuncRe = "^(__.*|_(init|start|fini)|(de)?register_tm_clones|call_gmon_start|frame_dummy|\.plt.*|_dl_relocate_static_pie)$";
+        needsMulti = "false";
+        stubRe = ''\bmain\b'';
+        stubText = "int main(void){return 0;/*stub provided by Compiler Explorer*/}";
+        supportsLibraryCodeFilter = "true";
+
+        libs = "";
+
+      };
+    in 
+    writeText "${compilerType}.defaults.properties" ''
+        ${attrToDot c_cpp }
+        ${ce_configuredCompilerNameAndBinaryListFromEntries "compiler" gccEntries}
+        ${ce_configuredCompilerNameAndBinaryListFromEntries "compiler" clangEntries}
+        tools=clangquery:lld:readelf:nm:strings:llvmdwarfdumpdefault
+      ''
+   );
+
+
+   cConfig = perform_kind true gccEntries clangEntries;
+   cppConfig = perform_kind false gccEntries clangEntries;
+
+
+in
+stdenv.mkDerivation {
+  pname = "compiler-explorer-cpp";
+  name = "Compiler Explorer GCC/CLANG";
```

Comment:
Name is used internally and isn't the right place for a human-readable description. We can add this in parentheses to meta.description.